Fix S-meter oscillation

argilo opened the pull request:
@LongnoseRob noticed in #994 that the S-meter sometimes begins oscillating. This can easily be reproduced by using a large buffer length with an RTL-SDR by setting the device string to `rtl=0,buflen=1048576`.

To fix this I've changed the rx_meter block to operate the same way the rx_fft block does: incoming samples are stored in a circular buffer (which can accommodate bursts up incoming samples up to one second long), and the same number of samples is always averaged. Clock measurements determine which batch of samples is averaged.

LongnoseRob commented:
@argilo Thank you very much for this fix.
I already built it on my sdr-system and will give it a few days of test time.
I think by the weekend we should have some feedback.

argilo commented:
@LongnoseRob Great, thanks!

I forgot to mention that this pull request includes some cleanup of the rx_meter block as well:

1. I've switched the detector type from RMS to average. Since it's measuring the signal power (and not amplitude) I don't think RMS is physically meaningful.
2. Removed the other detector types, which were never used.
3. Removed the `get_level` output, which was never used.
